tests/jquery.flot.composeImages.Test.js

```diff
@@ -0,0 +1,674 @@
+        'circle {' +
+            'stroke: black;' +
+            'stroke-width: 2px;' +
+        '}' +
+        '</style>' +
+        '<div id="test-container" style="width: 600px;height: 400px">' +
+        '<svg class="imgsrc" id="svgSource1" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg" width="100" height="100" title="svg1">' +
+            '<circle id="c1" cx="10" cy="10" r="5" style="fill:red"/>' +
+            '<circle id="c2" cx="30" cy="40" r="7" style="fill:#00FF00"/>' +
+            '<circle id="c3" cx="50" cy="70" r="9" style="fill:blue"/>' +
+        '</svg>' +
+        '<canvas class="imgsrc" id="canvasSource1" width="20" height="20" title="canvasSource1"></canvas>' +
+        '</div>' +
+        '<canvas id="myCanvas" width="150" height="150" style="border:1px solid #d3d3d3;"></canvas>'
+        ).find('.imgsrc').toArray();
+
+        var originalCanvas1 = document.getElementById("canvasSource1");
+        var destinationCanvas = document.getElementById("myCanvas");
+        var canvas1_Data; //used later
+        var canvas3_Data; //used later
+        var pixelData;//used later
+
+        function writeSomethingToCanvas(canvas, color) {
+            var ctx = canvas.getContext('2d');
+            ctx.rect(0, 0, 20, 20);
+            ctx.fillStyle = color;
+            ctx.fill();
+        }
+
+        writeSomethingToCanvas(originalCanvas1, "#FF0000");
+
+        expect(sources.length).toBe(2);
+
+        composeImages(sources, destinationCanvas).then(function() {
+            expect(destinationCanvas.width).toBe(100);
+            expect(destinationCanvas.height).toBe(100);
+
+            canvas1_Data = originalCanvas1.getContext('2d').getImageData(0, 0, 20, 20).data;
+            canvas3_Data = destinationCanvas.getContext('2d').getImageData(100 - 40 + 0, 0, 20, 20).data; //100 - 40 + 4
+            expect(matchPixelDataArrays(canvas1_Data, canvas3_Data)).toBe(true);
+
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(10, 10, 1, 1).data;
+            expect(matchPixelColor(pixelData, 255, 0, 0, 255)).toBe(true);
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(30, 40, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 255, 0, 255)).toBe(true);
+
+            pixelData = destinationCanvas.getContext('2d').getImageData(50, 70, 1, 1).data;
+            expect(matchPixelColor(pixelData, 0, 0, 255, 255)).toBe(true);
+
+            done();
+        }, null);
+    });
+
+    it('should call composeImages on one canvas and an SVG, which are totally overlapped with transparency. The SVG has a different size than the ones from other tests. One component of the SVG is partially outside of the view area.', function (done) {
+        var sources = placeholder.html('<style type="text/css">' +
+        '#canvasSource1 {position:relative; left:-180px; top:-10px;}' +
+        'circle {' +
+            'stroke: black;' +
+            'stroke-width: 4px;' +
+        '}' +
```

Comment:
In order to save more lines, the styles applied to the circle selector could be moved on the same line with the selector.

Reply:
done